runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5079,6 +5079,59 @@ class INTERPRETER_CLASS
 	}
 #endif /* JAVA_SPEC_VERSION >= 16 */
 
+#if JAVA_SPEC_VERSION >= 19
+	/* jdk.internal.vm.Continuation: private native boolean enter(); */
+	VMINLINE VM_BytecodeAction
+	enterContinuation(REGISTER_ARGS_LIST)
+	{
+		VM_BytecodeAction rc = EXECUTE_BYTECODE;
+		J9InternalVMFunctions *vmFuncs = _vm->internalVMFunctions;
+		jboolean res = JNI_FALSE;
+
+		j9object_t continuationObject = *(j9object_t*)_sp;
+
+		buildInternalNativeStackFrame(REGISTER_ARGS);
+		updateVMStruct(REGISTER_ARGS);
+
+		res = vmFuncs->enterContinuation(_currentThread, continuationObject);
+
+		VMStructHasBeenUpdated(REGISTER_ARGS);
+		if (VM_VMHelpers::exceptionPending(_currentThread)) {
+			rc = GOTO_THROW_CURRENT_EXCEPTION;
+			goto done;
+		}
+
+		restoreInternalNativeStackFrame(REGISTER_ARGS);
+		returnSingleFromINL(REGISTER_ARGS, res, 1);
+done:
+		return rc;
+	}
+
+	/* jdk.internal.vm.Continuation: private static native boolean yieldImpl(ContinuationScope scope); */
+	VMINLINE VM_BytecodeAction
+	yieldContinuation(REGISTER_ARGS_LIST)
+	{
+		VM_BytecodeAction rc = EXECUTE_BYTECODE;
+		J9InternalVMFunctions *vmFuncs = _vm->internalVMFunctions;
+		j9object_t continuationScope = *(j9object_t*)_sp;
+
+		updateVMStruct(REGISTER_ARGS);
+
```

Comment:
The continuation enter code (for resume) currently doesnt expect an INL frame. That being said, we should add one so the stack is walkable by the GC

Reply:
> The continuation enter code (for resume) currently doesnt expect an INL frame.

Ok. Because we're stacking a new interpreter entry on even continuation switch, this (and the enter bytecode action), aren't actually resuming on the continuation.  They stack an interpreter, run the continuation in it, and then restore the frame here.  Got it.

And yes, definitely need a frame here for the GC to walk it